code/modules/vending/_vending.dm

```diff
@@ -722,87 +856,134 @@ GLOBAL_LIST_EMPTY(vending_products)
 		return
 	switch(action)
 		if("vend")
-			if(!vend_ready)
-				return
-			if(panel_open)
-				to_chat(usr, "<span class='warning'>The vending machine cannot dispense products while its service panel is open!</span>")
-				return
-			vend_ready = FALSE //One thing at a time!!
-			var/datum/data/vending_product/R = locate(params["ref"])
-			var/list/record_to_check = product_records + coin_records
-			if(extended_inventory)
-				record_to_check = product_records + coin_records + hidden_records
-			if(!R || !istype(R) || !R.product_path)
-				vend_ready = TRUE
-				return
-			var/price_to_use = default_price
-			if(R.custom_price)
-				price_to_use = R.custom_price
-			if(R in hidden_records)
-				if(!extended_inventory)
-					vend_ready = TRUE
-					return
-			else if (!(R in record_to_check))
-				vend_ready = TRUE
-				message_admins("Vending machine exploit attempted by [ADMIN_LOOKUPFLW(usr)]!")
-				return
-			if (R.amount <= 0)
-				say("Sold out of [R.name].")
-				flick(icon_deny,src)
-				vend_ready = TRUE
-				return
-			if(onstation && ishuman(usr))
-				var/mob/living/carbon/human/H = usr
-				var/obj/item/card/id/C = H.get_idcard(TRUE)
+			. = vend(params)
+		if("select_colors")
+			. = select_colors(params)
 
-				if(!C)
-					say("No card found.")
-					flick(icon_deny,src)
-					vend_ready = TRUE
-					return
-				else if (!C.registered_account)
-					say("No account found.")
-					flick(icon_deny,src)
-					vend_ready = TRUE
-					return
-				var/datum/bank_account/account = C.registered_account
-				if(account.account_job && (account.active_departments & dept_req_for_free))
-					price_to_use = 0
-				if(coin_records.Find(R))
-					price_to_use = R.custom_premium_price ? R.custom_premium_price : extra_price
-				if(price_to_use && !account.adjust_money(-price_to_use))
-					say("You do not possess the funds to purchase [R.name].")
-					flick(icon_deny,src)
-					vend_ready = TRUE
-					return
+/obj/machinery/vending/proc/can_vend(user, silent=FALSE)
+	. = FALSE
+	if(!vend_ready)
+		return
+	if(panel_open)
+		to_chat(user, "<span class='warning'>The vending machine cannot dispense products while its service panel is open!</span>")
+		return
+	return TRUE
 
-				// each department (seller_department) will earn the profit
-				if(price_to_use && seller_department)
-					var/list/dept_list = SSeconomy.get_dept_id_by_bitflag(seller_department)
-					if(length(dept_list))
-						price_to_use = round(price_to_use/length(dept_list))
-						for(var/datum/bank_account/department/D in dept_list)
-							if(D)
-								D.adjust_money(price_to_use)
```

Comment:
you're removing the recent bank system change.

Reply:
yeah, it kept causing conflicts. I cut it out for the time being so I could work out the icon runtimes first